docs/source/chaincode4ade.rst

```diff
@@ -84,319 +61,479 @@ To keep things simple, let's use the following command:
-statements for the necessary dependencies for our chaincode. We'll import the
-chaincode shim package and the
-`peer protobuf package <https://godoc.org/github.com/hyperledger/fabric-protos-go/peer>`_.
-Next, let's add a struct ``SimpleAsset`` as a receiver for Chaincode shim functions.
+`fabric-contract-api interface <https://godoc.org/github.com/hyperledger/fabric-contract-api-go/contractapi>`_,
+so let's add the Go import statements for the necessary dependencies for our chaincode. We'll import the
+fabric contract api package and define our SmartContract.
 
 .. code:: go
 
-    package main
+  package main
 
-    import (
-    	"fmt"
+  import (
+    "fmt"
+    "log"
+    "github.com/hyperledger/fabric-contract-api-go/contractapi"
+  )
 
-    	"github.com/hyperledger/fabric-chaincode-go/shim"
-    	"github.com/hyperledger/fabric-protos-go/peer"
-    )
+  // SmartContract provides functions for managing an Asset
+  type SmartContract struct {
+  contractapi.Contract
+  }
 
-    // SimpleAsset implements a simple chaincode to manage an asset
-    type SimpleAsset struct {
-    }
+Next, let's add a struct ``Asset`` as a receiver for Chaincode functions.
+
+.. code:: go
+
+  // Asset describes basic details of what makes up a simple asset
+  type Asset struct {
+    ID             string `json:"ID"`
+    Color          string `json:"color"`
+    Size           int    `json:"size"`
+    Owner          string `json:"owner"`
+    AppraisedValue int    `json:"appraisedValue"`
+  }
 
 Initializing the Chaincode
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Next, we'll implement the ``Init`` function.
+Next, we'll implement the ``Init`` function to populate the ledger with some initial data.
 
 .. code:: go
 
-  // Init is called during chaincode instantiation to initialize any data.
-  func (t *SimpleAsset) Init(stub shim.ChaincodeStubInterface) peer.Response {
+  // InitLedger adds a base set of assets to the ledger
+  func (s *SmartContract) InitLedger(ctx contractapi.TransactionContextInterface) error {
+    assets := []Asset{
+      {ID: "asset1", Color: "blue", Size: 5, Owner: "Tomoko", AppraisedValue: 300},
+      {ID: "asset2", Color: "red", Size: 5, Owner: "Brad", AppraisedValue: 400},
+      {ID: "asset3", Color: "green", Size: 10, Owner: "Jin Soo", AppraisedValue: 500},
+      {ID: "asset4", Color: "yellow", Size: 10, Owner: "Max", AppraisedValue: 600},
+      {ID: "asset5", Color: "black", Size: 15, Owner: "Adriana", AppraisedValue: 700},
+      {ID: "asset6", Color: "white", Size: 15, Owner: "Michel", AppraisedValue: 800},
+    }
+
+    for _, asset := range assets {
+      assetJSON, err := json.Marshal(asset)
+      if err != nil {
+        return err
+      }
 
+      err = ctx.GetStub().PutState(asset.ID, assetJSON)
+      if err != nil {
+        return fmt.Errorf("failed to put to world state. %v", err)
+      }
+    }
+
+    return nil
   }
 
 .. note:: Note that chaincode upgrade also calls this function. When writing a
```

Comment:
This note should be removed. You no not need to call the Init function to upgrade the chaincode unless you use the Init required flag. If you are using the go contract API, you can use any function to initialize the chaincode.

The previous tutorial linked to this section here: https://hyperledger-fabric.readthedocs.io/en/latest/chaincode_lifecycle.html#step-three-approve-a-chaincode-definition-for-your-organization Given that we are trying to push people away from this feature, it may be best not to link, and let the feature be documented here.